src/simulation/deviceInterface/tempMeasurement/tempMeasurement.cpp

```diff
@@ -0,0 +1,123 @@
+/*
+ ISC License
+
+ Copyright (c) 2023, Autonomous Vehicle Systems Lab, University of Colorado Boulder
+
+ Permission to use, copy, modify, and/or distribute this software for any
+ purpose with or without fee is hereby granted, provided that the above
+ copyright notice and this permission notice appear in all copies.
+
+ THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+*/
+
+
+#include "tempMeasurement.h"
+#include <iostream>
+
+
+/*! This is the constructor for the module class.  It sets default variable
+    values and initializes the various parts of the model.
+    Don't allow random walk by default.
+ */
+TempMeasurement::TempMeasurement() : faultState{TEMP_FAULT_NOMINAL},
+                                    walkBounds{1E-15},
+                                    spikeProbability{0.1},
+                                    spikeAmount{2.0}
+{
+    this->noiseModel = GaussMarkov(1, this->RNGSeed);
+}
+
+TempMeasurement::~TempMeasurement() = default;
+
+/*! This method is used to reset the module and checks that required input messages are connected.
+    @return void
+*/
+void TempMeasurement::Reset(uint64_t CurrentSimNanos)
+{
+    if (!this->tempInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "TempMeasurement.tempInMsg was not linked.");
+    }
+
+    if (this->spikeProbability > 1.0 || this->spikeProbability < 0.0) {
+        bskLogger.bskLog(BSK_ERROR, "The probability of temperature spike on fault must be between 0 and 1.");
+    }
+    // set up gaussMarkov and random number generator parameters
+    this->spikeProbabilityGenerator.seed(this->RNGSeed);
+    this->noiseModel.setRNGSeed(this->RNGSeed);
+
+    Eigen::VectorXd nMatrix(1,1);
+    nMatrix(0,0) = this->senNoiseStd;
+    this->noiseModel.setNoiseMatrix(nMatrix);
+
+    Eigen::VectorXd pMatrix(1,1);
+    pMatrix(0,0) = 1.;
+    this->noiseModel.setPropMatrix(pMatrix);
+
+    Eigen::VectorXd bounds(1,1);
+    bounds(0,0) = this->walkBounds;
+    this->noiseModel.setUpperBounds(bounds);
+}
+
+
+void TempMeasurement::applySensorErrors()
```

Comment:
Add a method definition comment as you do with the other method.

Reply:
Done.